extensions/guacamole-auth-restrict/src/main/resources/controllers/timeRestrictionFieldController.js

```diff
@@ -0,0 +1,221 @@
+     */
+    $scope.weekDays = [
+        { id : '1', day : 'Monday' },
+        { id : '2', day : 'Tuesday' },
+        { id : '3', day : 'Wednesday' },
+        { id : '4', day : 'Thursday' },
+        { id : '5', day : 'Friday' },
+        { id : '6', day : 'Saturday' },
+        { id : '7', day : 'Sunday' },
+        { id : '*', day : 'All days' }
+    ];
+    
+    /**
+     * Remove the current entry from the list.
+     * 
+     * @param {TimeRestrictionEntry} entry
+     *     A restriction entry.
+     */
+    $scope.removeEntry = function removeEntry(entry) {
+        if (entry === null || entry.$$hashKey === '') {
+            return;
+        }
+        for (let i = 0; i < $scope.restrictions.length; i++) {
+            if ($scope.restrictions[i].$$hashKey === entry.$$hashKey) {
+                $scope.restrictions.splice(i,1);
+                return;
+            }
+        }
+    };
+    
+    /**
+     * Add an empty entry to the restriction list.
+     */
+    $scope.addEntry = function addEntry() {
+        $scope.restrictions.push(new TimeRestrictionEntry());
+    };
+    
+
+    /**
+     * Parse the provided string into an array containing the objects that
+     * represent each of entries that can then be displayed as a more
+     * user-friendly field.
+     * 
+     * @param {String} restrString
+     *     The string that contains the restrictions, un-parsed and as stored
+     *     in the underlying field.
+     *     
+     * @returns {TimeRestrictionEntry[]}
+     *     An array of objects that represents each of the entries as parsed
+     *     out of the string field, and which can be interpreted by the
+     *     AngularJS field for display.
+     */
+    const parseRestrictions = function parseRestrictions(restrString) {
+        
+        var restrictions = [];
+        
+        // If the string is null or empty, just return an empty array
+        if (restrString === null || restrString === "")
+            return restrictions;
+        
+        // Set up the RegEx and split the string using the separator.
+        const restrictionRegex = new RegExp('^([1-7*])(?::((?:[01][0-9]|2[0-3])[0-5][0-9])\-((?:[01][0-9]|2[0-3])[0-5][0-9]))$');
+        var restrArray = restrString.split(";");
+
+        // Loop through split string and process each item
+        for (let i = 0; i < restrArray.length; i++) {
+            
+            // Test if our regex matches
+            if (restrictionRegex.test(restrArray[i])) {
+                var currArray = restrArray[i].match(restrictionRegex);
+                var entry = new TimeRestrictionEntry();
+                entry.weekDay = '' + currArray[1];
+                entry.startTime = new Date(Date.UTC(1970, 1, 1, parseInt(currArray[2].slice(0,2)), parseInt(currArray[2].slice(2)), 0, 0));
+                entry.endTime = new Date(Date.UTC(1970, 1, 1, parseInt(currArray[3].slice(0,2)), parseInt(currArray[3].slice(2)), 0, 0));
```

Comment:
Apparently `Date.UTC()` does not deal with DST ("summer time") correctly:

```
var testDate1 = new Date();
undefined
testDate1.getHours();
16
testDate1.getTimezoneOffset();
240
testDate1.getUTCHours();
20
var testDate2 = new Date(Date.UTC(2023, 11, 3, 20, 00, 00, 00));
undefined
testDate2.getHours()
15
testDate2.getTimezoneOffset();
300
```

:face_with_spiral_eyes:

Reply:
I'm seeing consistent behavior on chrome regardless of whether I use `Date.UTC()` or not
```
new Date(Date.UTC(2023, 10, 1, 20, 00, 00, 00)).getTimezoneOffset();
420

new Date(2023, 10, 1, 20, 00, 00, 00).getTimezoneOffset();
420

new Date(Date.UTC(2023, 11, 1, 20, 00, 00, 00)).getTimezoneOffset();
480

new Date(2023, 11, 1, 20, 00, 00, 00).getTimezoneOffset();
480```

Reply:
It's also worth keeping in mind that the `Date()` and `Date.UTC()` constructors are zero-indexed for the month argument -- so `0` means January and `11` means December.

Reply:
> It's also worth keeping in mind that the Date() and Date.UTC() constructors are zero-indexed for the month argument -- so 0 means January and 11 means December.

Ah, yes, okay, that was at least part of the issue I was seeing. I've still got something going on in my code that isn't translating things quite correctly - the translation of the hour from local/browser time (the field) to the database is always off by an hour, and I've no idea why.

Reply:
Those `1970, 1, 1` arguments will throw things off by a month, and we're right at the cusp of a DST change so it makes sense that that could do it.

Reply:
Yeah, I realized that was causing issues and got rid of that, already, but I was still seeing some other oddities. Or maybe I just hadn't recompiled. Weird - seems to be working right now with changes I'll commit shortly.

Reply:
Okay, I think I've fixed that issue...

Reply:
While we're at it - why not use the current date instead of January, 1970?

There may have been changes in switchover dates for some locations for DST onset date since 1970 which may also screw up these comparisons.

Reply:
Yep, that's pretty much what I did - just pull the current date and use the year, month, and date from that, and then set the hour and minute from the configuration.